app/labor/badge_rewarder.rb

```diff
@@ -95,10 +96,11 @@ def self.award_streak_badge(num_weeks)
   end
 
   def self.award_badges(usernames, slug, message_markdown)
+    badge_id = Badge.find_by(slug: slug).id
```

Comment:
should this method just fail with `.find_by!` if the slug doesn't exist?

Reply:
It would fail with `NoMethodError: undefined method 'id' for nil:NilClass` if we can't find the slug,  so  `ActiveRecord::RecordNotFound: Couldn't find Badge` raised `.find_by!` seems preferable.

Reply:
I have replaced `find_by` with `find_by!` where appropriate.

Reply:
Thanks @lightalloy :-)